team-cards/src/index.js

```diff
@@ -1,7 +1,36 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
-import App from "./App.js";
+import { createBrowserRouter, RouterProvider, Outlet } from "react-router-dom";
+import Home from "./Components/Home.js";
+import ErrorComponent from "./Components/ErrorComponent.js";
+import Header from "./Components/Header.js";
+import UserDetails from "./Components/UserDetails.js";
+
+const AppLayout = () => (
+  <>
+    <Header />
+    <Outlet />
+  </>
+);
+
+const appRouter = createBrowserRouter([
+  {
+    path: "/",
+    element: <AppLayout />,
+    errorElement: <ErrorComponent />,
+    children: [
+      {
+        path: "/",
+        element: <Home />,
+      },
+      {
+        path: "/user/:id",
+        element: <UserDetails />,
```

Comment:
you're not passing props here so you can get rid of props argument from ctor. same for home component.